### ISSUE TITLE
`@override`n method should inherit signature

### ISSUE THREAD

**Issue by toth-dev** [astral-sh/ty#3102]
### Summary

When you override an type annotated method with a method without annotation, I think it should inherit the signature from the original method. This would be considered an error:

```python
from abc import ABC, abstractmethod
from typing import override

class A(ABC):
    @abstractmethod
    def f(self) -> int: ...

class B(A):
    @override
    def f(self):
        return 'f'
```

**Comment by dhruvmanila** [comment]
I'm not sure if ty should raise this error via the logic of inheriting the signature for an `override` but https://github.com/astral-sh/ty/issues/128 should at least resolve this case by inferring the return type of `f` as `Literal["f"]` which should then raise an error. Other type checkers also emit an error using return type inference.

**Comment by carljm** [comment]
Thanks for the report! I don't think any type checker implements this in precisely this way (making `return 'f'` an error by assuming an inherited return type), but some type checkers do error on this, via inference the other direction: they infer a return type of `str` for `B.f` and then consider it an invalid override of `A.f`. I think that when we fix #128, the latter behavior will fall out naturally.
